src/libraries/Microsoft.PowerFx.Core/Parser/ParserOptions.cs

```diff
@@ -52,6 +52,11 @@ public class ParserOptions
         /// </summary>
         internal bool AllowParseAsTypeLiteral { get; set; }
 
+        /// <summary>
+        /// Flag to allow single user-definitons to be parsed without semicolon in the end.
+        /// </summary>
+        public bool AllowSingleUserDefinition { get; set; }

```

Comment:
>AllowSingleUserDefinition [](http://example.com/codeflow?start=20&length=25)

Is the intention this flag is always true?

Reply:
I don't have a strong opinion on this but we need to have this today as PowerApps requires semicolon. I wish this was like AllowChainExpressions in which `;` is optional . For this to be eventually true / have  `AllowChainExpressions` like behavior here , we need to confirm that this is fine with PowerApps and bit more UserDefinition parser changes

Reply:
But changing from error (semi is require) to non-error (it's not required) is not a breaking change...